### DISCUSSION TITLE
How do I add additional custom preference in the request body.

### DISCUSSION THREAD

**Discussion by kashaudhan** [appbaseio/reactivesearch#2004] [Q&A]
I'm working on a reactive native app which requires the request body to like :
`{"preference":"results"}
{"query":{"bool":{"must":[{"bool":{"should":[{"geo_distance":{"distance":"2km","location":"10.057, 70.99"}}],"minimum_should_match":1}}]}},"size":10,"_source":{"includes":["*"],"excludes":[]},"from":0}
{"preference":"googleMap"}
{"query":{"bool":{"must":[{"bool":{"should":[{"geo_distance":{"distance":"2km","location":"13.0353557, 77.59878739999999"}}],"minimum_should_match":1}}]}},"size":10000,"from":0}`

insted of:
`
{"preference":"results"}
{"query":{"bool":{"must":[{"bool":{"should":[{"geo_distance":{"distance":"2km","location":"10.057, 70.99"}}],"minimum_should_match":1}}]}},"size":10,"_source":{"includes":["*"],"excludes":[]},"from":0}
`

I want the request body to have two preferences(one with `result` & other with `googleMap`).

I'm using `@appbaseio/reactivesearch-native@0.12.0` to connect the search ui with the elastic search backend.

Please help me out if you know the way to get is done. 
Thanks.

**Comment by kashaudhan** [comment]
@siddharthlatest could you please help me out.

**Reply by siddharthlatest** [reply]
@kashaudhan reactivesearch-native is deprecated in favor of [react-native-searchbox](https://docs.appbase.io/docs/reactivesearch/react-native-searchbox/quickstart/).
